src/styles/main.scss

```diff
@@ -1,3 +1,72 @@
+$cell-size: 100px;
+
+html {
+  font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
+}
+
 body {
   margin: 0;
 }
+
+.field {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $cell-size * 7 + 6px;
+  border: 1px solid #000;
+  padding: 10px;
+  gap: 1px;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    font-size: 30px;
+    box-sizing: border-box;
+    width: $cell-size;
+    height: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    transition-duration: 0.5s;
+  }
+
+  &__day:hover {
+    transform: translateY(-20px);
+    background-color: #ffc0cb;
+    cursor: pointer;
+  }
+}
+
+@for $day from 1 through 31 {
+  .calendar :nth-child(n + #{$day})::after {
+    content: "#{$day}" ;
+  }
+}
+
+@for $day from 28 to 31 {
```

Comment:
Add a modifier for the month length to the .calendar selector, e.g. &--month-length--#{day}.